### PR TITLE
[OV] Optimized compression to MXFP4 data type

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
### Changes

Added optimized compression to MXFP4 data type for OpenVINO backend.

| Model                | Memory Before (MiB) | Memory After (MiB)         | Time Before (sec) | Time After (sec)         |
|----------------------|---------------------|----------------------------|-------------------|--------------------------|
| llama-3.2-1b bf16    | 2778.55             | 2548.37 (-8.29%)           | 66.33             | 24.95 (-62.40%)          |
| llama-3.2-1b fp16    | 3434.61             | 2963.03 (-13.73%)          | 62.12             | 24.85 (-59.98%)          |
| llama-3.2-1b fp32    | 2041.79             | 1576.43 (-22.81%)          | 62.72             | 25.85 (-58.77%)          |
|                      |                     |                            |                   |                          |
| phi4-mini bf16       | 6384.81             | 5725.66 (-10.33%)          | 197.36            | 66.22 (-66.44%)          |
| phi4-mini fp16       | 8863.53             | 8375.75 (-5.51%)           | 195.85            | 66.93 (-65.82%)          |
| phi4-mini fp32       | 4406.82             | 3897.91 (-11.54%)          | 195.25            | 68.83 (-64.76%)          |
|                      |                     |                            |                   |                          |
| llama-3.1-8b bf16    | 7297.72             | 6096.06 (-16.46%)          | 413.86            | 135.25 (-67.32%)         |
| llama-3.1-8b fp16    | 7946.93             | 8311.89 (+4.58%)           | 413.64            | 136.05 (-67.11%)         |
| llama-3.1-8b fp32    | 7310.48             | 5043.89 (-30.03%)          | 411.45            | 140.66 (-65.81%)         |

### Reason for changes

Improving user experience.

### Related tickets

164717

### Tests

Extended optimized compression tests.

### CODE REVIEW

File: src/nncf/version.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = "2.18.0"
+__version__ = "2.18.0.dev0+66c0366aedirty"
```
Comment (nikita-savelyevv) [inline src/nncf/version.py]:
```suggestion
__version__ = "2.18.0"
```

File: src/nncf/quantization/algorithms/weight_compression/weight_lowering.py [hunk 2]
```diff
@@ -508,17 +515,30 @@ def integer_quantize_dequantize_weight(
...
+    indexes = fns.searchsorted(quantile_centers, norm_weight)
+    quantiles = fns.from_numpy(quantiles_np, backend=indexes.backend)
+
+    if mode == CompressWeightsMode.MXFP4:
+        # If in-between two quantiles, round to the nearest even quantile.
+        shifted_indexes = fns.clip(indexes + 1, 0, quantiles.size - 1)
+        dist_left = fns.abs(norm_weight - quantiles[indexes])
+        dist_right = fns.abs(norm_weight - quantiles[shifted_indexes])
+        choose_right = (dist_right < dist_left) | ((dist_left == dist_right) & ((shifted_indexes + 1) % 2 == 0))
+        indexes = fns.where(choose_right, shifted_indexes, indexes)
```
Comment (ljaljushkin) [inline src/nncf/quantization/algorithms/weight_compression/weight_lowering.py:538]: what about adding small unit test for such rounding?
maybe also add some corner cases: -0.0, 0.0, more than max, less then min

```python
@pytest.mark.parametrize(
        "center_idx,expected_idx",
        [
            (0, 1),
            (1, 1),
            (2, 3),
            (6, 7),
            (7, 7),
            (-1, -2),
        ],
    )
    def test_exact_quantile_center_values(center_idx, expected_idx, description):
        """Test that values exactly at quantile centers round to nearest even index."""
        center_val = CENTER_OF_MXFP4_QUANTILES[center_idx]
        expected_q = MXFP4_QUANTILES[expected_idx]
        
        norm_weight = Tensor(np.array([center_val], dtype=np.float32))
        result = _calculate_float_quantized_weight(norm_weight, CompressWeightsMode.MXFP4)
        
        # Verify the result matches expected quantile
        assert result.data[0] == expected_q, (
            f"{description}: Expected {expected_q}, got {result.data[0]} "
            f"for center value {center_val}"
        )
```

File: src/nncf/openvino/optimized_functions/functions.py [hunk 3]
```diff
@@ -107,17 +107,17 @@ def do_float_quantization(
     precomputed_scale: Optional[Tensor] = None,
 ) -> tuple[Tensor, Tensor, Tensor]:
     """
-    Computes quantization scale if not provided, and performs corresponding nf4 weight quantization.
+    Computes quantization scale if not provided, and performs corresponding float weight quantization.
     For NF4 quantization quantizes the weights to 16 levels on [-1, 1] interval.
-    TODO(nikita-savelyevv): add support for MXFP4 and MXFP8_E4M3 once ticket 164851 is resolved
+    For MXFP4 quantization quantizes the weights to 16 levels on [-6, 6] interval.
```
Comment (ljaljushkin) [inline src/nncf/openvino/optimized_functions/functions.py]:
```suggestion
   NF4 format uses 16 levels in [-1, 1] range, while MXFP4 uses 16 levels in [-6, 6].
```

File: tests/cross_fw/test_templates/template_test_nncf_tensor.py [hunk 4]
```diff
@@ -98,6 +100,25 @@ def test_operator_clone(self):
...
+        fn = BOOLEAN_OPERATOR_MAP[op_name]
+        res = fn(tensor_a, value)
+        res_nncf = fn(nncf_tensor_a, value)
+
+        assert res.dtype == res_nncf.data.dtype
+        assert all(res == res_nncf.data)
+        assert isinstance(res_nncf, Tensor)
+        if (
+            self.backend() != TensorBackend.tf
+        ):  # native Tensorflow operaors do not guarantee to return a tensor on an initial device.
```
Comment (ljaljushkin) [inline tests/cross_fw/test_templates/template_test_nncf_tensor.py]:
```suggestion
        ):  # native Tensorflow operators do not guarantee to return a tensor on an initial device.
```

File: src/nncf/openvino/optimized_functions/functions.py (hunk 3 above)
Comment (nikita-savelyevv) [reply src/nncf/openvino/optimized_functions/functions.py]: Done

File: tests/cross_fw/test_templates/template_test_nncf_tensor.py (hunk 4 above)
Comment (nikita-savelyevv) [reply tests/cross_fw/test_templates/template_test_nncf_tensor.py]: Done

File: src/nncf/quantization/algorithms/weight_compression/weight_lowering.py (hunk 2 above)
Comment (nikita-savelyevv) [reply src/nncf/quantization/algorithms/weight_compression/weight_lowering.py:538]: Added. Also, `tests/openvino/optimized_functions/test_compression_functions.py::test_quantization_alignment` fails without the highlighted `if` logic.
